### ISSUE TITLE
Per object permissions for mutation with an input type?

### ISSUE THREAD

**Issue by Mapiarz** [blb-ventures/strawberry-django-plus#189]
Hi! I would like to add a per object permission to one of my mutations. The mutation accepts an input type (not a partial) with `booking_id` field. I need to resolve that node and check the permission for it.

When checking the condition, I only have access to `GraphQLResolveInfo` which is not enough to call `GlobalID.resolve_node`.

Any ideas or suggestions?

**Comment by bellini666** [comment]
Hey @Mapiarz ,

Why do you say that you only have access to `GraphQLResolveInfo`? If you mean the `info` parameter that you can receive at the mutation resolver, that is actually an `Info` object. 

They share similar attributes, but when the field/mutation is a django one, you can access the request at `info.context.request` (and thus, retrieve the current user by `info.context.request.user`, if that's what you need)

**Comment by Mapiarz** [comment]
Hi @bellini666 ,
I wasn't specific enough. I wanted to handle this in a directive resolver, before the actual mutation resolver is executed. As far as I can tell, in the directive resolver I only have `GraphQLResolveInfo` and access to input params is not strongly typed (it's just a dict, not my input type but that I can live with).

`HasObjPerm` directive is promising, but it only supports checking the permission for the root or return node. In my case, resolution of the return node is not that simple and cannot be done automatically and thus I cannot use this approach.

In short, I was hoping I could create a directive in which I can get a global id from the input (which I can already do via the args dict), resolve the node (which I don't know how to do) and then check a given permission on that object instance.

For the time being, I'm doing this permission check inline in the mutation resolver, exactly as you suggested.

**Comment by bellini666** [comment]
@Mapiarz oh, I see what you mean now.

TBH, there's an initial implementation that might work, but even though I implemented it I'm actually not using it in my personal projects because I think it is "too magical" (there isn't actually a link between the directive and the argument) and I want to try another approach (but still haven't had the time to do that)

You can use [get_with_perms](https://github.com/blb-ventures/strawberry-django-plus/blob/main/strawberry_django_plus/permissions.py#L209) instead of `id.resolve_node`. It receives a `GlobalID`/`ID` and will run it through all `HasObjPerm` defined in the directives of the field, raising `PermissionDenied` if any of them fails.
